packages/core/src/checker/checker-facade.ts

```diff
@@ -34,73 +35,67 @@ export class CheckerFacade {
     checkResult$: Observable<MutantResult>;
     passedMutant$: Observable<MutantRunPlan>;
   } {
-    if (!this.options.checkers.length) {
-      return {
-        checkResult$: EMPTY,
-        passedMutant$: input$,
-      };
-    }
-
-    const checkResult$ = new Subject<MutantResult>();
-    let previousPassedMutants$ = input$;
-
-    for (let i = 0; i < this.options.checkers.length; i++) {
-      const passedMutants$ = new Subject<MutantRunPlan>();
-      this.executeChecker(i, previousPassedMutants$, checkResult$, passedMutants$);
-      previousPassedMutants$ = passedMutants$;
+    let checkResult$: Observable<MutantResult> = EMPTY;
+    let passedMutant$ = input$;
+    for (const checkerName of this.options.checkers) {
+      // Use this checker
+      const { failedCheckResult$, passedCheckResult$ } = this.executeChecker(checkerName, passedMutant$);
+
+      // Prepare for the next one
+      passedMutant$ = passedCheckResult$;
+      checkResult$ = concat(
+        checkResult$,
+        failedCheckResult$.pipe(map((failedMutant) => this.mutationTestReportHelper.reportCheckFailed(failedMutant.mutant, failedMutant.checkResult)))
+      );
     }
-
-    previousPassedMutants$.subscribe({
-      complete: () => {
-        this.log.debug('Checker(s) finished.');
-
-        checkResult$.complete();
-
-        this.checkerPool.dispose();
-        this.concurrencyTokenProvider.freeCheckers();
-      },
-    });
-
     return {
-      checkResult$: checkResult$.asObservable(),
-      passedMutant$: previousPassedMutants$,
+      checkResult$,
+      passedMutant$: passedMutant$.pipe(
+        tap({
+          complete: () => {
+            this.checkerPool.dispose();
+            this.concurrencyTokenProvider.freeCheckers();
+          },
+        })
+      ),
     };
   }
 
-  private async executeChecker(
-    checkerIndex: number,
-    previousPassedMutants$: Observable<MutantRunPlan>,
-    checkResult$: Subject<MutantResult>,
-    passedMutant$: Subject<MutantRunPlan>
-  ) {
-    const mutants = await lastValueFrom(merge(previousPassedMutants$).pipe(toArray()));
-
-    const groups = await firstValueFrom(
-      this.checkerPool.schedule(of(0), async (checker) => {
-        const mutantArray = mutants.map((m) => m.mutant);
-        const group = await checker.createGroups?.(checkerIndex, mutantArray);
-
-        return group ?? mutants.map((m) => [m.mutant]);
+  private executeChecker(checkerName: string, input$: Observable<MutantRunPlan>) {
+    const group$ = this.checkerPool
+      .schedule(input$.pipe(bufferTime(CHECK_BUFFER_MS)), async (checker, mutants) => {
```

Comment:
I'm using `bufferTime` here, that makes it so the result of the previous checker can be used in the next one.

Reply:
I doubt if this is preferred over the previous solution. If the checkerpool is still full with tasks of the previous checker we don't need to add new tasks because all the "cpu cores" are still busy. This way we are creating new tasks with inefficient groups because there are probably not a lot of mutants to be grouped while this new group has to wait to be executed by the checkerpool. I think we want to create groups when the checkerpool is (almost) done with the previous checker, because there are more mutants ready for the next checker so we can create larger groups. We should maybe discuss this in a meeting so I can explain it better and we could consider the consequences of both implementations.

Reply:
As discussed in person. I doubt it would be a big performance gain. It is largely a theoretical discussion anyway until we get a second checker implementation.